File: testcases/benchmark/searching/binary_search.f90
function binary_search(arr, n, target) result(idx)
    integer, intent(in) :: arr(:), n, target
    integer :: idx, low, high, mid
    low = 1
    high = n
    idx = -1
    do while (low <= high)
        mid = low + (high - low) / 2
        if (arr(mid) == target) then
            idx = mid
            return
        else if (arr(mid) < target) then
            low = mid + 1
        else
            high = mid - 1
        end if
    end do
end function binary_search

program main
end program main

File: testcases/benchmark/searching/binary_search.rs
fn binary_search(arr: &[i32], target: i32) -> i32 {
    let mut low: i32 = 0;
    let mut high: i32 = arr.len() as i32 - 1;
    while low <= high {
        let mid = low + (high - low) / 2;
        if arr[mid as usize] == target {
            return mid;
        } else if arr[mid as usize] < target {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    -1
}

fn main() {}
